yoyodyne/evaluators.py

```diff
@@ -1,28 +1,36 @@
 """Evaluators."""
 
 from __future__ import annotations
+import abc
+import argparse
 import dataclasses
+from typing import List
 
+import numpy
 import torch
 from torch.nn import functional
 
+# from torchmetrics.text import CharErrorRate
+from torchmetrics.functional.text.helper import _edit_distance
```

Comment:
It occurs to me that this is protected and we're using it anyways we ought to idk, alias it, or just use our own, or use a different third-party one? I have this one lying around which uses a numpy array instead of Python lists if you want it. 

```python
def _edit_distance(x: Labels, y: Labels) -> int:
    idim = len(x) + 1
    jdim = len(y) + 1
    table = numpy.zeros((idim, jdim), dtype=numpy.uint16)
    table[1:, 0] = 1
    table[0, 1:] = 1
    for i in range(1, idim):
        for j in range(1, jdim):
            if x[i - 1] == y[j - 1]:
                table[i][j] = table[i - 1][j - 1]
            else:
                c1 = table[i - 1][j]
                c2 = table[i][j - 1]
                c3 = table[i - 1][j - 1]
                table[i][j] = min(c1, c2, c3) + 1
    return int(table[-1][-1])
```

but don't hold the PR up for just that.

Reply:
I can test it really quick?

Reply:
Sure. You can turn it up to `uint32` if you expect strings longer than 65k.

Reply:
That's worrisome if only because i've been using this basic implementation for a very long time and in many places...

Reply:
Their lines 340-343 are different than my:

```python
table[1:, 0] = 1
table[0, 1:] = 1
```

And I think theirs might be right...that should be increasing.

Reply:
Sorry, have some meetings now but will return to this in an hour or two.

Reply:
Ok as a sanity check, if I do this instead:

```
table = numpy.zeros((idim, jdim), dtype=numpy.uint16)
        # table[1:, 0] = 1
        # table[0, 1:] = 1
        for i in range(idim):
            table[i][0] = i
        for j in range(jdim):
            table[0][j] = j
```

I reproduce the results. Off the top of my head I think this one is right, and your implementation might be for LCS? I will try to find references...

Reply:
Yes, I think your diagnosis is correct. It also occurred to me that you could use `empty` rather than `zeros` that way. I put it into a gist: 
https://gist.github.com/kylebgorman/14cc4e238bc6d80784df4511c29b3d55.